src/grid/ctrl.coffee

```diff
@@ -53,37 +61,46 @@ class GridCtrl
   ###
   init: ->
     @_headView
-      .setTags( null, 'TH' )
       .setTarget( @element.find( @HEADER_ELEMENT ) )
+      .setTags( null, 'TH' )
     @_rowsView
       .setTarget( @element.find( @BODY_ELEMENT ) )
+    @_pagination
+      .on( @_pagination.CHANGE, @updateRows )
+      .init()
+    @_datasource = new DatasourceArray()
+      .data( @_config.data )
 
-    datasource = new GridDatasourceArray()
-    datasource.data( @_config.data )
-    @setDatasource( datasource )
-
-    return @update()
+    # Count is obtain to config pagination
+    @_datasource.count()
+      .then ( count ) =>
+        @_pagination.setCount( count )
 
   ###
-   * Updates element with data rows
+   * Updates grid element with data rows
    * @return {Promise}
   ###
-  update: ->
+  updateRows: =>
+    if @_config.pagination isnt false
+      @_datasource
+        .limit( @_pagination.getLimit() )
+        .skip( @_pagination.getSkipped() )
+
     @_datasource
       .keys( @_head.keys() )
       .get()
       .then ( data ) =>
         @_rowsView.setRows( data )
         @_headView.setRows( [ @_head.labels() ] )
-        return data
 
-  ###*
+  ###
    * Destroys component and removes added DOM elements
    * @return {this}
```

Comment:
I am not sure this notation is valid. I feel more comfortable with @return {Object} this
Can we check it somehow?